src/webapp/src/characters/charactersSlice.ts

```diff
@@ -0,0 +1,83 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+interface ApiExtraInformation {
+  count: number;
+  pages: number;
+  next: string;
+  prev: string | null;
+}
+
+interface Character {
+  id: number;
+  name: string;
+  status: string;
+  species: string;
+  type: string;
+  gender: string;
+  origin: {
+    name: string;
+    url: string;
+  };
+  location: {
+    name: string;
+    url: string;
+  };
+  episode: string[];
+  url: string;
+  created: Date;
+  image: string;
+}
+
+export interface ApiData {
+  results: Character[];
+  info: ApiExtraInformation;
+  error?: string | null;
+}
+
+const initialState = {
+  results: [] as Character[],
+  info: {} as ApiExtraInformation,
+  error: null,
+} as ApiData;
+
+interface ApiKnownError {
+  error: string;
+}
+
+export const getCharacters = createAsyncThunk<
+  ApiData,
+  void,
+  { rejectValue: ApiKnownError }
+>('characters/getCharacters', async (_, { rejectWithValue }) => {
+  const response = await fetch(`https://rickandmortyapi.com/api/character`);
+
+  if (response.status === 404) {
+    return rejectWithValue((await response.json()) as ApiKnownError);
+  }
+
+  return (await response.json()) as ApiData;
+});
+
+const charactersSlice = createSlice({
+  name: 'characters',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(getCharacters.pending, () => {});
+
+    builder.addCase(getCharacters.rejected, (state, action) => {
+      if (action.payload) {
+        return { ...state, error: action.payload.error };
+      }
+      return { ...state, error: action.error.message };
+    });
+
+    builder.addCase(getCharacters.fulfilled, (state, { payload }) => ({
+      ...state,
+      results: payload.results,
```

Comment:
In the same way, as the other comment, if useful information from this `results` object is only `{ name, species, image, id }` then we want to store in redux or anywhere, only those. The rest is noise we dont need :)

Reply:
Also very true, I tend to store that kind of info *in case we need it* (crazy of me, I know!). I see how it is noise to have it there in the meantime. 

I removed everything we don't need. I like how the `store` looks now, much better. Thanks for pointing in it out.